src/lib/operators/join_hash.cpp

```diff
@@ -21,54 +21,76 @@
 
 namespace opossum {
 
+bool JoinHash::supports(JoinMode join_mode, PredicateCondition predicate_condition, DataType left_data_type,
```

Comment:
What's the crucial benefit with declaring the `supports`-function as static? Since all `AbstractJoinOperator`-derived classes implement this function, `supports(…)` could be implemented in `AbstractJoinOperator` as (non-static) pure virtual function to guarantee that possible upcoming derived classes have implemented this function.
Imo `supports` is in the same semantic level as `name()` of `AbstractOperator` but for join operators.

Reply:
nothings "crucial" ;)

The reasons I choose static over virtual are:
* With static there is no need to build a join operator that we never use; no need to build a join operator with a configuration that it doesn't support. Not (temporarily) constructing invalid state is what's probably most important to me here.
* Implementation of supports() is sufficiently (implicitly) enforced by places calling `JoinOperator::supports(...)` - so there is no realtistic way you could add a new join operator (which is added to the JoinTestRunner) without implementing supports()
* You'd save passing join_mode, secondary_predicates and predicate_condition with virtual, granted. But you'd not get around to pass left/right_data_type.